spec/lib/tasks/location_children_count_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rake'
+
+RSpec.describe 'location:fix_children_count' do
+  before do
+    Rails.application.load_tasks
+    @parent = create(:location)
+    @child1 = create(:location, parent: @parent)
+    @child2 = create(:location, parent: @parent)
+    @grandchild = create(:location, parent: @child1)
+
+    # Manually set children_count to 0 for all locations
+    # rubocop:disable Rails/SkipsModelValidations
```

Comment:
Can you set the children_count on creation instead?

```
@parent = create(:location, children_count: 0)
@child1 = create(:location, parent: @parent, children_count: 0)
@child2 = create(:location, parent: @parent, children_count: 0)
@grandchild = create(:location, parent: @child1, children_count: 0)
```

This might not work as ancestry might override it each time.

Reply:
Might also be worth having a children_count that is set to 100 (or some other non zero value) and confirm that also gets corrected.